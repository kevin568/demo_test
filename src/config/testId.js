export const TEST_ID = {
  TODO_INPUT: "TODO_INPUT",
  TODO_ITEM: "TODO_ITEM",
  ADD_BUTTON: "ADD_BUTTON",
  APP_TITLE: "APP_TITLE",
};
