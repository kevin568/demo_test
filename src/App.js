import logo from "./logo.svg";
import "./App.css";
import react, { useState } from "react";
import { Container, Title } from "./Style.css";
import { Control } from "./component/Control";
import { Content } from "./component/Content";
import { TEST_ID } from "./config/testId";

function App() {
  const [list, setList] = useState([]);
  return (
    <Container>
      <Title data-testid={TEST_ID.APP_TITLE}>Todo List Sample</Title>
      <Control list={list} setList={setList} />
      <Content list={list} setList={setList} />
    </Container>
  );
}

export default App;
