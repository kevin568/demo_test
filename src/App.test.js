import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import { TEST_ID } from "./config/testId";

describe("Todo List", () => {
  test("App container should be rendered", () => {
    const { container } = render(<App />);
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
  test("App title should be rendered", () => {
    render(<App />);
    const apptitle = screen.getByTestId(TEST_ID.APP_TITLE);
    expect(apptitle).toHaveTextContent("Todo List Sample");
  });
  test("Todo input should be rendered", () => {
    render(<App />);
    const todoInput = screen.getByTestId(TEST_ID.TODO_INPUT);
    const addButton = screen.getByTestId(TEST_ID.ADD_BUTTON);

    fireEvent.change(todoInput, { target: { value: "test" } });
    fireEvent.click(addButton);

    const todoItem = screen.getByTestId(`${TEST_ID.TODO_ITEM}_test`);
    expect(todoItem).toBeInTheDocument();
  });
  test("Todo item should be removed", () => {
    render(<App />);
    const todoInput = screen.getByTestId(TEST_ID.TODO_INPUT);
    const addButton = screen.getByTestId(TEST_ID.ADD_BUTTON);

    fireEvent.change(todoInput, { target: { value: "delete" } });
    fireEvent.click(addButton);

    const todoItem = screen.getByTestId(`${TEST_ID.TODO_ITEM}_delete`);
    fireEvent.click(todoItem);
    expect(todoItem).not.toBeInTheDocument();
  });
});
