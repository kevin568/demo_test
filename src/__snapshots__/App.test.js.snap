// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo List App container should be rendered 1`] = `
<div
  class="sc-beyTiQ cVCinz"
>
  <div
    class="sc-hLQTFJ dKSoTB"
    data-testid="APP_TITLE"
  >
    Todo List Sample
  </div>
  <div
    class="sc-guDMob irPtzy"
  >
    <input
      class="sc-dmyDmy fLldPA"
      data-testid="TODO_INPUT"
      value=""
    />
    <button
      class="sc-eDLKEg jhIUPb"
      data-testid="ADD_BUTTON"
    >
      Add
    </button>
  </div>
  <div
    class="sc-guDMob irPtzy"
  >
    <ul
      class="sc-jTQDJr kPNYHK"
    />
  </div>
</div>
`;
